babel.config.js

```diff
@@ -1,6 +1,99 @@
-module.exports = {
-  presets: [
-    '@babel/preset-react',
-    '@babel/preset-env'
-  ]
-};
+module.exports = function(api) {
+  var validEnv = ['development', 'test', 'production']
+  var currentEnv = api.env()
+  var isDevelopmentEnv = api.env('development')
+  var isProductionEnv = api.env('production')
+  var isTestEnv = api.env('test')
+
+  if (!validEnv.includes(currentEnv)) {
+    throw new Error(
+      'Please specify a valid `NODE_ENV` or ' +
+        '`BABEL_ENV` environment variables. Valid values are "development", ' +
+        '"test", and "production". Instead, received: ' +
+        JSON.stringify(currentEnv) +
+        '.'
+    )
+  }
+
+  return {
+    presets: [
+      isTestEnv && [
+        '@babel/preset-env',
+        {
+          targets: {
+            node: 'current'
+          },
+          modules: 'commonjs'
+        },
+        '@babel/preset-react'
+      ],
```

Comment:
- [ ] please have a look at [this totorial](https://betterprogramming.pub/react-with-rails-2022-bd28e1fcd355#ad0a) the babel.config.js file should look a lot different and the webpack.config.js file is missing, please make sure to implement these correctly